pkg/pillar/cmd/volumemgr/blob.go

```diff
@@ -536,6 +536,17 @@ func populateInitBlobStatus(ctx *volumemgrContext) {
 			log.Functionf("populateInitBlobStatus: blob %s in CAS could not get mediaType", blobInfo.Digest)
 			continue
 		}
+		// In Kubevirt eve, we share same user containerd repository between eve and k3s containers.
+		// So volumemgr should ignore publishing blobs which are not downloaded through eve download mechanism
+		// In /run/volumemgr/BlobStatus we should only see the blobs downloaded by eve.
+		if ctx.hvTypeKube {
```

Comment:
This code publishes the blobs present in containerd repository. We need to explicitly check for hvTypeKube for couple of reasons.

1) kubevirt eve will have blobs downloaded by k3s which are not eve downloaded. So we need to filter those away and not publish those. Else they get GC in  gcBlobStatus()
2) non-kubevirt EVE will also have blobs which does not have eve-downloaded label ie all the images created before this PR is merged, in other words all customer installs. 


This PR is only setting the label when the image is created.  BTW, I am thinking about another followup PR where in non-kubevirt eve we can blindly set the eve-downloaded=true label on all images and blobs when volumemgr starts. I did not include that in this PR because its cleaner and will be easily back portable if we want to do it.

Reply:
> non-kubevirt EVE will also have blobs which does not have eve-downloaded label ie all the images created before this PR is merged, in other words all customer installs

That is a good point. We should get past that as rapidly as we can, to keep things simple. Things tend to linger in large complex projects like this. :-)

>  I am thinking about another followup PR where in non-kubevirt eve we can blindly set the eve-downloaded=true label on all images and blobs when volumemgr starts

Yes! And then after that, can we eliminate all of it, so that it ends up uniform?

Reply:
Not sure what you mean by eliminate all of it.  Eventually all images downloaded by eve will have eve-downloaded label. Yes we can do GC or publishBlobStatus only if that label is present.

Reply:
Yup that's what I meant.

Reply:
> That is a good point. We should get past that as rapidly as we can, to keep things simple. Things tend to linger in large complex projects like this. :-)

@deitch note that a customer might have had EVE download some content in 2020 and still be running the same device in 2030. So unless we also do something in EVE which sets the label in containerd for all existing images, then we can't rely on using the label for kvm/xen/etc.